src/Std/Classes/Ord.lean

```diff
@@ -295,6 +295,13 @@ class LawfulBEqCmp {α : Type u} [BEq α] (cmp : α → α → Ordering) : Prop
   /-- If two values compare equal, then they are logically equal. -/
   compare_eq_iff_beq {a b : α} : cmp a b = .eq ↔ a == b
 
+theorem LawfulBEqCmp.not_compare_eq_iff_beq_eq_false {α : Type u} [BEq α] {cmp}
+    [LawfulBEqCmp (α := α) cmp] {a b : α} : ¬ cmp a b = .eq ↔ (a == b) = false := by
+  rw [Bool.eq_false_iff, ne_eq, not_congr]
+  exact compare_eq_iff_beq
+
+export LawfulBEqCmp (compare_eq_iff_beq not_compare_eq_iff_beq_eq_false)
```

Comment:
I wonder whether these lemmas -- and `compare_eq_iff_eq` as well -- should rather be called `cmp_eq_iff_eq`. Then `compare_eq_iff_*` could refer to the `Ord`-instance-based variants.

Reply:
Yes, this sounds like a good idea. Another option could be to not export these lemmas.